.github/workflows/test.yml

```diff
@@ -100,28 +97,86 @@ jobs:
-        AWS_SECRET_ACCESS_KEY: ${{ secrets.DELTA_ACCESS_KEY }}
         REGISTRY_USERNAME: ${{ secrets.REGISTRY_USERNAME }}
         REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
       run: |
         docker login -u $REGISTRY_USERNAME -p $REGISTRY_PASSWORD quay.io
         cdk deploy --require-approval never --outputs-file outputs.json
+        $(jq -r ".[].ekskubeconfigcmd" outputs.json) --kubeconfig ./kubeconfig
+    - name: Collect diagnostic data
+      if: always() && (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
+      env:
+        KUBECONFIG: ./kubeconfig
+        LOG_DIR: /tmp/k8s-cluster-state/cdk
+      run: |
+        set +e
+        curl -Lo /usr/local/bin/kubectl  "https://dl.k8s.io/release/v1.23.6/bin/linux/amd64/kubectl"
+        for ns in domino-platform domino-compute domino-system kube-system; do
+          mkdir -p $LOG_DIR/$ns
+          kubectl -n $ns get ing -o yaml > $LOG_DIR/$ns/ingress.txt
+          kubectl -n $ns get po -o yaml > $LOG_DIR/$ns/pods.txt
+          kubectl -n $ns describe po > $LOG_DIR/$ns/pods-described.txt
+          kubectl -n $ns get pvc -o yaml > $LOG_DIR/$ns/pvcs.txt
+          kubectl -n $ns get svc -o yaml > $LOG_DIR/$ns/svcs.txt
+          kubectl -n $ns describe svc > $LOG_DIR/$ns/svcs-described.txt
+          kubectl -n $ns get events > $LOG_DIR/$ns/events.txt
+        done
+        kubectl get pv -o yaml > $LOG_DIR/pvs.txt
+        kubectl get no -o yaml > $LOG_DIR/nodes.txt
+        kubectl describe no > $LOG_DIR/nodes-described.txt
+    - name: Setup cloudformation-only user for safe stack deletion
+      if: (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
+      working-directory: ./convert/cloudformation-only
+      run: |
+        echo "{\"region\":\"us-west-2\",\"tags\":{},\"suffix\":\"$NAME\"}" > terraform.tfvars.json
+        terraform init
+        terraform plan -out=terraform.plan
+        terraform apply -auto-approve terraform.plan
+    - name: Setup terraform-aws-eks conversion process
+      if: (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
+      working-directory: ./convert
+      run: |
+        pip install -r requirements.txt
+        ssh-keygen -t rsa -f dummy.pem -N ''
+        ./convert.py print-stack --stack-name $NAME --region $AWS_REGION --verbose --yaml > stack-data.yaml
+        ./convert.py create-tfvars --stack-name $NAME --region $AWS_REGION --ssh-key-path ./dummy.pem > terraform/terraform.tfvars.json
+        ./convert.py get-imports --stack-name $NAME --region $AWS_REGION > terraform/imports.sh
+    - name: Run terraform
+      if: (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
+      working-directory: ./convert/terraform
+      run: |
+        terraform init
+        bash imports.sh
+        terraform plan -out=terraform.plan
```

Comment:
Wonder if it's worth checking for resource destruction (vs. add/change) and failing in here if there are any?

Reply:
Good call, I added a thing.